src/types/utils.ts

```diff
@@ -328,8 +329,16 @@ export function isFunctionCallExternal(call: FunctionCall): boolean {
         return true;
     }
 
-    if (call.vExpression.typeString.includes(FunctionVisibility.External)) {
-        return true;
+    const exprT = inference.typeOf(call.vExpression);
+
+    if (exprT instanceof FunctionType) {
+        if (exprT.visibility === FunctionVisibility.External) {
+            return true;
+        }
+
+        if (exprT.visibility === FunctionVisibility.Public) {
+            return call.vExpression.getChildrenByType(MemberAccess, true).length > 0;
```

Comment:
Do we handle this case correctly on this line:

1. x.functionName, where x is a variable of some type, T, and we have a `using Lib for T;` in the code

Reply:
This is a good catch. I see that we mark such things with `implicitFirstArg`. It seems to have only use for `using Lib for T` is used. I will try to filter out those and add related test case.

Reply:
Addressed in a160ce9